packages/x-tree-view/src/useTreeItem2/useTreeItem2.ts

```diff
@@ -132,6 +167,13 @@ export const useTreeItem2 = <TPlugins extends DefaultTreeViewPlugins = DefaultTr
       onFocus: createRootHandleFocus(externalEventHandlers),
       onBlur: createRootHandleBlur(externalEventHandlers),
       onKeyDown: createRootHandleKeyDown(externalEventHandlers),
+      ...(itemsReordering?.enabled
+        ? {
+            draggable: instance.canItemBeDragged(itemId),
+            onDragStart: createRootHandleDragStart(externalEventHandlers),
+            onDragEnd: createRootHandleDragEnd(externalEventHandlers),
```

Comment:
```suggestion
            onDragEnd: createRootHandleDragEnd(externalEventHandlers),
            onDragOver: (event) => event.preventDefault(),
```

To prevent draggable items from returning to their original position:


| Before | After |
| ------- | ----- |
| <video src="https://github.com/mui/mui-x/assets/13808724/73716e27-3d91-4fd7-a6b6-56c8e3688cd7" /> | <video src="https://github.com/mui/mui-x/assets/13808724/80a46f98-eb75-48f6-bdf7-d2c97de15b49" /> |

Reply:
That's interesting, I can't reproduce your bug
Are you just dragging and then dropping in another location?

Reply:
> Are you just dragging and then dropping in another location?

Yes. I can reproduce it in Chromium 123, as well as in Safari 17.4.1 on MacOS 14.4.1